FrontEnd/src/components/ProfileDetail/MainInfo/TitleInfo.jsx

```diff
@@ -0,0 +1,171 @@
+import { useState, useMemo } from 'react';
+import useSWR from 'swr';
+import { Badge } from 'antd';
+import { StarOutlined, StarFilled } from '@ant-design/icons';
+import { PropTypes } from 'prop-types';
+import useSWRMutation from 'swr/mutation';
+
+import DefaultLogo from './DefaultLogo';
+import classes from './TitleInfo.module.css';
+
+function TitleInfo({ isAuthorized, data }) {
+  const [isSaved, setIsSaved] = useState(data.is_saved);
+  const profile = useMemo(() => {
+    return {
+      id: data.id,
+      personId: data.person,
+      name: data.name,
+      activities: !data.activities.length
+        ? null
+        : data.activities.map((activity) => activity.name).join(', '),
+      region: data.region_display ? data.region_display : '',
+      categories:
+        data.categories.length > 4
+          ? data.categories.slice(0, 4)
+          : data.categories,
+      isSaved: data.is_saved,
+    };
+  }, [data]);
+
+  // TODO: change the logic of getting user.id when PR with hooks will be merged
+
+  const authToken = localStorage.getItem('Token');
+  const { data: userData } = useSWR(
+    authToken ? `${process.env.REACT_APP_BASE_API_URL}/api/auth/users/me/` : null,
+    url =>
+        fetch(url, {
+            method: 'GET',
+            headers: {
+                'Authorization': `Token ${authToken}`,
+            },
+        }).then(res => res.json()),
```

Comment:
we need also handle promise rejection and probably show toast error message

Reply:
This code was deleted as useUser hook added.